mlflow/sagemaker/__init__.py

```diff
@@ -1656,6 +1659,204 @@ def _does_model_exist(model_name, sage_client):
         return True if response else False
 
 
+class SageMakerPlugin(BaseDeploymentClient):
```

Comment:
I had to register the module that contains the SageMaker client (see https://github.com/mlflow/mlflow/pull/4971/commits/059d64f3f77f17f802da3692a6d16a7e5a6c50c4) because the plugin manager requires some functions are defined at the module level when initializing a client object:
https://github.com/mlflow/mlflow/blob/9faf3b04c9f6b3766a64d7433dd3e85bcdf27fff/mlflow/deployments/plugin_manager.py#L93-L94